content/learn/vp_agof.adoc

```diff
@@ -0,0 +1,307 @@
+
+== Ansible framework methods
+
+The main methods for setting up an Ansible framework are as follows:
+
+[NOTE]
+====
+The focus here is primarily on the AWS-based installation method. 
+====
+
+=== AWS based install
+
+This method is ideal for organizations deploying AAP on AWS infrastructure. In AAP 2.4, the default installation process uses AWS and offers a fully automated setup. It requires AWS credentials and builds an AWS image with Red Hat’s ImageBuilder, setting up AAP within an AWS VPC and subnet. The installation program creates all necessary resources, including AAP Controllers and, optionally, additional components such as the Automation Hub.
+
+This is the easiest method if you already use AWS, as it automates the provisioning of resources, including VMs and network configurations. This requires AWS infrastructure and credentials.
+
+=== Custom Ansible controller (API install)
+
+In this method, you provide an existing Ansible Automation Platform (AAP) Controller endpoint, either on bare metal or in a private cloud, without needing AWS or pre-configured VMs. You specify the manifest, endpoint hostname, admin credentials, and pass the installation process to a predefined `controller_config_dir`. This is suitable for complex or custom topologies where you want full control over the deployment.
+
+This method provides maximum flexibility and is designed for advanced users who have their own AAP installations, either on-prem or in complex environments that do not fit into the default or AWS-centric model. You need an existing AAP controller, which might not be ideal for users new to AAP or those looking for more hands-off installation
+
+== Creating a validated pattern using the AGOF framework
+
+To deploy a validated pattern, follow these steps:
+
+. Clone the link:https://github.com/mhjacks/agof_demo_config[Ansible GitOps Framework Minimal Demo] repository. This is a minimal pattern that demonstrates how to use the Ansible GitOps Framework.
++
+[source,terminal]
+----
+$ git clone https://github.com/validatedpatterns-demos/agof_minimal_demo.git
+----
+
+. Clone the link:https://github.com/validatedpatterns/agof[AGOF] repository. This serves as the provisioner for the pattern
++
+[source,terminal]
+----
+$ git clone https://github.com/validatedpatterns-demos/agof_minimal_demo.git
+----
+
+=== Deploying using the AWS-based install method
+
+This method: 
+
+. Builds an AWS image using Red Hat's ImageBuilder.
+
+. Deploys that image onto a new AWS VPC and subnet.
+
+. Deploys AAP on that image using the command line installer. 
+
+. Hands over the configuration of the AAP installation to the specified `controller_config_dir`. 
```

Comment:
In AGOF this is now  `agof_controller_config_dir` because passing `controller_config_dir` to the AAP installer would do something else (specifically, it would use that directory _on the AAP controller node_.

Reply:
I see that the variable references were still incorrect in AGOF's readme. I'm fixing those now, sorry.

Reply:
https://github.com/validatedpatterns/agof/pull/28